### PR TITLE
feat(test): add dts testing

### PR DESCRIPTION
**Pull request by antfu**
Added some of the dts testing (ref and watch) copied from vue-next. Will try to adopting more in the future.

This PR also updates Github Actions to using yarn and triggers dts tests.

### CODE REVIEW

**Comment by pikax** [comment]
Can we hold this for a bit, tests are about to change in vue-next
https://github.com/vuejs/vue-next/pull/1106

**Comment by antfu** [comment]
Didn't notice that, and sure!

**Comment by pikax** [comment]
PR merged

**Comment by antfu** [comment]
Test script updated.

File: test-dts/ref.test-d.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  expectType<boolean>(falseRef.value);
+
+  // ref true
+  const trueRef = ref<true>(true);
+  expectType<Ref<true>>(trueRef);
+  expectType<true>(trueRef.value);
+
+  // tuple
+  // Not supported by Vue 2
+  // expectType<[number, string]>(unref(ref([1, '1'])));
```
Comment (pikax) [inline test-dts/ref.test-d.ts:35]: This should be supported 🤔

Comment (antfu) [reply test-dts/ref.test-d.ts]: Oh. You are right, I forget to pull the head. Thanks

File: .github/workflows/nodejs.yml [hunk 2]
```diff
@@ -1,29 +1,35 @@
...
-        npm run build --if-present
-        npm test
-      env:
-        CI: true
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install
+        run: yarn install
```
Comment (pikax) [inline .github/workflows/nodejs.yml:22]:
```suggestion
        run: yarn install --frozen-lockfile
```

Comment (antfu) [reply .github/workflows/nodejs.yml]: Didn't know that, thanks!

**Pull request merged**